frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx

```diff
@@ -83,20 +91,26 @@ const ResourceSection: React.FC<ResourceSectionProps> = ({ flags }) => {
     return options;
   }, [invalidTypes, canIncludeKnative, t]);
 
-  return (
-    <FormSection title={t('devconsole~Resource type')} fullWidth>
-      <div>{t('devconsole~Select the resource type to generate')}</div>
-      <SelectInputField
-        name={fieldName}
-        options={selectInputOptions}
-        variant={SelectVariant.single}
-        onChange={onChange}
-        getLabelFromValue={(value: string) => t(ReadableResourcesNames[value])}
-        hideClearButton
-        toggleOnSelection
-      />
-    </FormSection>
-  );
+  if (
+    !['edit', 'knatify'].includes(values.formType) &&
+    values.import?.selectedStrategy?.type !== ImportStrategy.SERVERLESS_FUNCTION
+  ) {
```

Comment:
It's just an opinion and I'm fine with this as well. But you need fewer intentions if you return null if the conditions aren't matched.

```suggestion
  if (
    ['edit', 'knatify'].includes(values.formType) ||
    values.import?.selectedStrategy?.type === ImportStrategy.SERVERLESS_FUNCTION
  ) {
    return null;
  }
```